central/administration/events/datastore/telemetry.go

```diff
@@ -0,0 +1,88 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/administration/events"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/telemetry/phonehome"
+)
+
+// Gather the number of administrative events
+func Gather(ds DataStore) phonehome.GatherFunc {
+	return func(ctx context.Context) (map[string]any, error) {
+		ctx = sac.WithGlobalAccessScopeChecker(ctx,
+			sac.AllowFixedScopes(
+				sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+				sac.ResourceScopeKeys(resources.Administration),
+			),
+		)
+		props := map[string]any{}
+		errorList := errorhelpers.NewErrorList("Administrative Events Telemetry")
+		err := phonehome.AddTotal(ctx, props, "Administrative Events", func(ctx context.Context) (int, error) {
+			return ds.CountEvents(ctx, search.EmptyQuery())
+		})
+		errorList.AddError(err)
+		err = phonehome.AddTotal(ctx, props, "Info type Administrative Events", func(ctx context.Context) (int, error) {
+			return ds.CountEvents(ctx,
+				search.NewQueryBuilder().
+					AddStrings(search.EventLevel, storage.AdministrationEventLevel_ADMINISTRATION_EVENT_LEVEL_INFO.String()).
+					ProtoQuery(),
+			)
+		})
+		errorList.AddError(err)
+		err = phonehome.AddTotal(ctx, props, "Warning type Administrative Events", func(ctx context.Context) (int, error) {
+			return ds.CountEvents(ctx,
+				search.NewQueryBuilder().
+					AddStrings(search.EventLevel, storage.AdministrationEventLevel_ADMINISTRATION_EVENT_LEVEL_WARNING.String()).
+					ProtoQuery(),
+			)
+		})
```

Comment:
Extract function refactoring suggestion:

```go
func countAdmEvents(ds DataStore, label, value string) func (context.Context) (int, error) {
	return func(ctx context.Context) (int, error) {
		return ds.CountEvents(ctx, search.NewQueryBuilder().
				AddStrings(label, value).
				ProtoQuery())
	}
}
```
So that:
```go
errorList.AddError(phonehome.AddTotal(ctx, props, "Warning type Administrative Events",
	countAdmEvents(ds, search.EventLevel, storage.AdministrationEventLevel_ADMINISTRATION_EVENT_LEVEL_WARNING.String()))
```

Reply:
Done, let me know what you think